processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -651,5 +663,12 @@ func (c *WatchClient) extractNamespaceLabelsAnnotations() bool {
 }
 
 func needContainerAttributes(rules ExtractionRules) bool {
-	return rules.ContainerImageName || rules.ContainerImageTag || rules.ContainerID
+	switch {
+	case rules.ContainerImageName,
+		rules.ContainerName,
+		rules.ContainerImageTag,
+		rules.ContainerID:
+		return true
+	}
+	return false
```

Comment:
Not sure I agree with this style. The former version seems more readable to me :)